store.go

```diff
@@ -1237,41 +1238,36 @@ func (s *store) allLayerStores() ([]roLayerStore, error) {
 }
 
 // readAllLayerStores processes allLayerStores() in order:
-// It locks the store for reading, checks for updates, and calls
-//
-//	(data, done, err) := fn(store)
-//
-// until the callback returns done == true, and returns the data from the callback.
-//
-// If reading any layer store fails, it immediately returns ({}, true, err).
+// It locks the store for reading, checks for updates, and iterates through (store, nil).
 //
-// If all layer stores are processed without setting done == true, it returns ({}, false, nil).
+// If reading any layer store fails, it yields (nil, err) and terminates.
 //
 // Typical usage:
 //
-//	if res, done, err := s.readAllLayerStores(store, func(…) {
-//		…
-//	}; done {
-//		return res, err
+//	for rlstore, err := range s.readAllLayerStores() {
```

Comment:
Note to self: This looks almost _too easy_, without warning callers about the semantics. Maybe `range iterateLockingAllLayerStores` ?

Reply:
`iterateReadLockingAllLayerStores`? As usual, I tend towards unwieldy names.